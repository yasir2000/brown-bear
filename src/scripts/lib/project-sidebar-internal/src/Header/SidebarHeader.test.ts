/**
 * Copyright (c) 2021-Present Enalean
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

import { shallowMount } from "@vue/test-utils";
import SidebarHeader from "./SidebarHeader.vue";
import { SIDEBAR_CONFIGURATION } from "../injection-symbols";
import { example_config } from "../project-sidebar-example-config";
import { ref } from "vue";

describe("SidebarHeader", () => {
    it("displays the sidebar header", () => {
        const wrapper = shallowMount(SidebarHeader, {
            global: {
                provide: {
                    [SIDEBAR_CONFIGURATION.valueOf()]: ref(example_config),
                },
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("does not display the administration link when the user is not a project administrator", () => {
        const config = example_config;
        config.user.is_project_administrator = false;
        const wrapper = shallowMount(SidebarHeader, {
            global: {
                provide: {
                    [SIDEBAR_CONFIGURATION.valueOf()]: ref(config),
                },
            },
        });

        expect(wrapper.find("[data-test=project-administration-link]").exists()).toBe(false);
    });
});
