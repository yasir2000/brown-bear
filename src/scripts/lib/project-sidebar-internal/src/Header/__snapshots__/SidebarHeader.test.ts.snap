// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarHeader displays the sidebar header 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="project-sidebar-header-name"
  >
    <div
      class="project-title-container"
    >
      <a
        class="project-sidebar-title"
        data-test="project-sidebar-title"
        href="/projects/project1"
      >
        <span
          aria-hidden="true"
          class="project-sidebar-title-icon"
        >
          ðŸŒ·
        </span>
         project1
      </a>
      <span
        class="project-title-spacer"
      />
      <a
        class="project-administration-link"
        data-test="project-administration-link"
        href="https://myinstance.example.com/project/admin/?group_id=999"
      >
        <i
          class="fas fa-cog project-administration-link-icon"
          title="Project administration"
        />
      </a>
    </div>
    <privacy-badge-stub />
  </div>
  <project-announcement-stub />
  <project-flags-stub />
  <linked-projects-stub />
  
</div>
`;
